test/test_nested_tensor_masking.py

```diff
@@ -3,42 +3,309 @@
+    def test_gen_nested_tensor(self):
+        nt = utils.gen_nested_tensor(seed=1, nested_dim=1, tensor_dim=1, size_low=2, size_high=2)
+        self.assertTrue(NT.is_nested_tensor(nt))
+        self.assertEqual(nt.nested_dim(), 1)
+        self.assertEqual(nt.size(), (2, 2))
+        self.assertEqual(nt[0].dim(), 1)
+        self.assertEqual(nt[0].size(), torch.Size([2]))
 
-    def test_simple(self):
-        values = [torch.rand(i) for i in range(10)]
-        random.shuffle(values)
-        nt = nestedtensor.nested_tensor(values)
-        tensor, mask = nt.to_tensor_mask()
+        nt = utils.gen_nested_tensor(seed=1, nested_dim=2, tensor_dim=1, size_low=2, size_high=2)
+        self.assertTrue(NT.is_nested_tensor(nt))
+        self.assertEqual(nt.nested_dim(), 2)
+        self.assertEqual(nt.size(), (2, 2, 2))
+        self.assertEqual(nt[0].dim(), 2)
+        self.assertEqual(nt[0].size(), (2, 2))
+        self.assertEqual(nt[0][0].size(), torch.Size([2]))
+
+        nt = utils.gen_nested_tensor(seed=1, nested_dim=2, tensor_dim=2, size_low=2, size_high=2)
+        self.assertTrue(NT.is_nested_tensor(nt))
+        self.assertEqual(nt.nested_dim(), 2)
+        self.assertEqual(nt.size(), (2, 2, 2, 2))
+        self.assertEqual(nt[0].dim(), 3)
+        self.assertEqual(nt[0].size(), (2, 2, 2))
+        self.assertEqual(nt[0][0].size(), torch.Size([2, 2]))
+        self.assertEqual(nt[0][0][0].size(), torch.Size([2]))
+
+        nt = utils.gen_nested_tensor(seed=1, nested_dim=2, tensor_dim=2, size_low=10, size_high=10)
+        self.assertTrue(NT.is_nested_tensor(nt))
+        self.assertEqual(nt.nested_dim(), 2)
+        self.assertEqual(nt.size(), (10, 10, 10, 10))
+        self.assertEqual(nt[0].dim(), 3)
+        self.assertEqual(nt[0].size(), (10, 10, 10))
+        self.assertEqual(nt[0][0].size(), torch.Size([10, 10]))
+        self.assertEqual(nt[0][0][0].size(), torch.Size([10]))
 
-    def test_nested(self):
-        nt = utils.gen_nested_tensor(2, 2, 2)
+    def test_to_tensor_mask(self):
+        nt = utils.gen_nested_tensor(1, 1, 1, size_low=2, size_high=2)
         tensor, mask = nt.to_tensor_mask()
+        self.assertEqual(mask, torch.tensor(True))
+        self.assertEqual(mask.size(), torch.Size([]))
+        self.assertEqual(mask.dim(), 0)
+        self.assertEqual(tensor.size(), torch.Size([2, 2]))
+        self.assertEqual(tensor.dim(), 2)
+        self.assertEqual(tensor.dim(), nt.dim())
 
-    def test_tensor_mask(self):
-        nt = utils.gen_nested_tensor(2, 2, 2, size_low=1, size_high=2)
+        nt = NT.nested_tensor([
+            NT.nested_tensor([
+                NT.nested_tensor([
+                    torch.tensor([1, 2])
+                ]),
+            ]),
+            NT.nested_tensor([
+                NT.nested_tensor([
+                    torch.tensor([3, 4]),
+                    torch.tensor([5, 6])
+                ])
+            ])
+        ])
+        
+        # mask_dim = None
```

Comment:
I'd split up all these subcases into separate tests such as "test_to_tensor_mask_dim_1" or whatever is most descriptive. That'll make it easier to figure out what failed and what's covered. If you wanted to be even more descriptive you'd add some more comments on the purpose of each test.

Reply:
Yeah, I was thinking about it. Agree, its cleaner. I think comments would be a bit overkill. Will add them where its not obvious what is being tested or what are the expectations.